File: solutions/typescript/reverse-string/1/reverse-string.ts
/*
 * input: string "abc" 
 * output: string "cba"
 */

export function reverse(input: string): string {
  const chars = [...input]
  let swap = ""
  for (let i = 0; i < chars.length / 2; i++) {
    swap = chars[i]
    chars[i] = chars[chars.length - i - 1]
    chars[chars.length - i - 1] = swap
  }
  return chars.join('')
}

File: solutions/csharp/reverse-string/1/ReverseString.cs
using System;
using System.Linq;

public static class ReverseString
{
    public static string Reverse(string input)
    {
        var arr = input.ToArray();
        for (int i = 0; i < arr.Length / 2; i++)
        {
            var swap = arr[i];
            arr[i] = arr[^(i + 1)];
            arr[^(i + 1)] = swap;
        }
        return new String(arr);
    }
}
